examples/advanced/docker/nvflare_with_docker.ipynb

```diff
@@ -0,0 +1,429 @@
+   "outputs": [],
+   "source": [
+    "! ./workspace/example_docker_project/prod_00/mylocalhost/startup/docker.sh -d"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e6aa064e-91c6-4b4d-bc09-3b29ae504ca7",
+   "metadata": {},
+   "source": [
+    "To check that the docker image has started, run:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "84d8c51c-966f-4d1b-b3a5-077bfc18a595",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! docker ps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a3af85d-4885-462a-90ce-fbf27fc2aa45",
+   "metadata": {},
+   "source": [
+    "### Docker Run Command for Clients\n",
+    "The following cell will show the contents of the `docker.sh` script for the FL client site-1, the default name for the first client configured in the `project.yml` for this example. If you have changed the name of the FL clients, please replace site-1 with your FL client name below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "512a256c-55af-4d53-8619-f2eef52c1f63",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! cat workspace/example_docker_project/prod_00/site-1/startup/docker.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58a609cf-c534-4d6d-a4ab-7002f717bbc4",
+   "metadata": {},
+   "source": [
+    "Much of this is the same as the `docker.sh` script for launching the Docker container for FL server. The Docker image is set to the same image we specified in `project.yml`. \n",
+    "\n",
+    "Again, `/workspace` is mapped to the directory that is the parent of the one containing this `docker.sh` script, and it is used for the working directory with the `-w` option.\n",
+    "\n",
+    "By default, `NETARG` is set to add the option `--net=host`, but since clients do not need to open any ports this is not needed as long as the FL client is able to connect to the FL server.\n",
+    "\n",
+    "The Docker script for FL clients has the additional variable `MY_DATA_DIR` set to `/home/flcient/data` by default which is mapped to `/data` in the container. You can set this to a custom value by running `export MY_DATA_DIR=$SOME_DIRECTORY` before `docker.sh`.\n",
+    "\n",
+    "The `GPU2USE` variable is also added to keep track of the option for what GPUs to use for the container. Uncomment a line setting the value of `GPU2USE` to use GPUs: `--gpus=all` to use all available GPUs, `--gpus=2` to use two GPUs, and `--gpus=\"device=0,2\"` for specifying specific GPUs where the numbers after `device=` are the GPU IDs.\n",
+    "\n",
+    "If you run the `docker.sh` script with the `-d` flag for detached mode, the `docker run` command that is executed will launch the Docker image in detached mode and automatically start the FL client. Otherwise, the container will start in interactive mode so you can manually start the client with the `start.sh` command in the startup directory by typing `./startup/start.sh`.\n",
```

Comment:
To be able to use "--gpu" flag, they might need to install nvidia container toolkit "https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#installing-the-nvidia-container-toolkit"

Reply:
Can you add a note point to this^

Reply:
added